dpc/src/transaction/transaction.rs

```diff
@@ -199,6 +204,27 @@ impl<N: Network> Transaction<N> {
             return false;
         }
 
+        let input_public_variables: Vec<_> = self
+            .transitions
+            .iter()
+            .flat_map(|t| t.input_public_variables(self.ledger_root, transitions.root()))
+            .collect();
+
+        let output_public_variables: Vec<_> =
+            self.transitions.iter().flat_map(|t| t.output_public_variables()).collect();
+
+        match self.kernel_proof.verify(&input_public_variables, &output_public_variables) {
```

Comment:
This is the only place that verifies the new batch proofs. Is there any other method that needs to be updated?

Reply:
This is the only place! I've tried to keep the surfaces as deduped as possible